main.ts

```diff
@@ -51,22 +68,31 @@ bot.chatType(["group", "supergroup"]).filter((ctx) =>
     await ctx.reply("Could not resolve file path.");
     return;
   }
-  const formData = new FormData();
-  formData.set("repository", env.REPOSITORY_CLONE_URL);
-  formData.set("branch", branch);
-  formData.set(
-    "url",
-    new URL(`/file/${encodeURIComponent(file_path)}`, env.BASE_URL).href,
-  );
-  res = await fetch(env.PATCH_PUSHER_API_URL, {
-    method: "POST",
-    body: formData,
-  });
-  if (res.status == 200) {
+  try {
+    await applyFile(branch, file_path);
     await ctx.reply("Patch submitted.");
-  } else {
-    console.log(res.status);
-    await ctx.reply("Failed to submit the patch.");
+  } catch (err) {
+    await ctx.reply(err instanceof Error ? err.message : String(err));
+  }
+});
+
+bot.command("apply", async (ctx) => {
+  const branch = ctx.msg.text.split(/\s/)[1];
```

Comment:
`ctx.match`?

Reply:
Wouldn’t that need a `bot.on(`?

Reply:
https://grammy.dev/guide/commands.html#arguments

Reply:
Oy. Firmware upgrade available.